aptos-move/framework/move-stdlib/src/natives/bcs.rs

```diff
@@ -74,13 +84,52 @@ fn native_to_bytes(
     Ok(smallvec![Value::vector_u8(serialized_value)])
 }
 
+/***************************************************************************************************
+ * native fun serialized_size
+ *
+ *   gas cost: size_of(output)
+ *
+ *   If the getting the type layout or serialization results in error, a special failure
+ *   cost is charged.
+ *
+ **************************************************************************************************/
+fn native_serialized_size(
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    let (val, layout) = process_bcs_args(context, ty_args, args)?;
+
+    // Note:
+    // This function reuses the same gas costs as bcs::to_bytes. We might want to
+    // consider to charge less if
+    //   1) calculation of serialized size changes, or
+    //   2) the final gas cost (it is u64, not vector like in bcs::to_bytes).
+    // Charging for failure during layout construction can stay the same.
+    let serialized_size = match serialized_size_allowing_delayed_values(&val, &layout) {
+        Some(serialized_size) => serialized_size as u64,
+        None => {
+            context.charge(BCS_TO_BYTES_FAILURE)?;
+            return Err(SafeNativeError::Abort {
+                abort_code: NFE_BCS_SERIALIZATION_FAILURE,
+            });
+        },
+    };
+    context.charge(BCS_TO_BYTES_PER_BYTE_SERIALIZED * NumBytes::new(serialized_size))?;
```

Comment:
let's add new params for gas charging, even if they are initialized to same values

Reply:
done